src/components/checkbox/GroupableCheckboxConnected.tsx

```diff
@@ -0,0 +1,44 @@
+import { connect } from 'react-redux';
+import * as _ from 'underscore';
+import { IReactVaporState } from '../../ReactVapor';
+import { IReduxAction, ReduxUtils } from '../../utils/ReduxUtils';
+import { IInputDispatchProps, IInputProps, IInputStateProps } from '../input/Input';
+import {Checkbox} from './Checkbox';
+import {ICheckboxState} from './CheckboxReducers';
+import {
+  addGroupedCheckbox, IGroupableCheckboxActionPayload, removeGroupedCheckbox,
+  toggleGroupedCheckbox,
+} from './GroupableCheckboxActions';
+import {IGroupableCheckboxesState} from './GroupableCheckboxReducers';
+
+export interface IGroupableCheckboxOwnProps extends IInputProps {
+  isParent?: boolean;
+  parentId?: string;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IGroupableCheckboxOwnProps): IInputStateProps => {
+  const groupableCheckboxesState: IGroupableCheckboxesState = _.findWhere(state.groupableCheckboxes, { parentId: ownProps.parentId || ownProps.id });
+  if (groupableCheckboxesState) {
+    const checkbox: ICheckboxState = ownProps.isParent ? groupableCheckboxesState.parent : _.findWhere(groupableCheckboxesState.checkboxes, {id: ownProps.id});
+    return {
+      checked: checkbox && checkbox.checked,
+      indeterminate: ownProps.isParent && groupableCheckboxesState.nbChecked > 0 &&
+                     groupableCheckboxesState.nbChecked < groupableCheckboxesState.total,
+    };
+  }
+
+  return {
+    checked: ownProps.checked,
```

Comment:
this is a bit weird that you use the ownProps.checked here, in theory it should be undefined since you are in a connected component, could this do the same

```
const mapStateToProps = (state: IReactVaporState, ownProps: IGroupableCheckboxOwnProps): IInputStateProps => {
    const groupableCheckboxesState: IGroupableCheckboxesState = _.findWhere(state.groupableCheckboxes, {parentId: ownProps.parentId || ownProps.id});
    let checkbox: ICheckboxState;
    if (groupableCheckboxesState) {
        checkbox = ownProps.isParent
            ? groupableCheckboxesState.parent
            : _.findWhere(groupableCheckboxesState.checkboxes, {id: ownProps.id});
    }

    return {
        checked: checkbox && checkbox.checked,
        indeterminate: ownProps.isParent
            && groupableCheckboxesState.nbChecked > 0
            && groupableCheckboxesState.nbChecked < groupableCheckboxesState.total,
    };
};  
```

Reply:
updated, I use defaultChecked props for the first render